ros_buildfarm/common.py

```diff
@@ -172,7 +172,10 @@ def get_distribution_repository_keys(urls, key_files):
 def get_binary_package_versions(apt_cache, debian_pkg_names):
     versions = {}
     for debian_pkg_name in debian_pkg_names:
-        pkg = apt_cache[debian_pkg_name]
+        pkg = apt_cache.get(debian_pkg_name)
+        if not pkg:
+            prov = apt_cache.get_providing_packages(debian_pkg_name)
+            pkg = apt_cache[prov[0]]  # Just grab the first one
```

Comment:
Does defaulting to the first available also match apt's behavior? Is it possible that we're introducing some non-determinism here since python dicts are unordered? I know that `prov` here is a list, but it's potentially formed by reading in dicts if I was to speculate on the apt cache implementation.